src/platform_impl/windows/event_loop.rs

```diff
@@ -135,17 +135,38 @@ static GET_POINTER_TOUCH_INFO: Lazy<Option<GetPointerTouchInfo>> =
 static GET_POINTER_PEN_INFO: Lazy<Option<GetPointerPenInfo>> =
     Lazy::new(|| get_function!("user32.dll", GetPointerPenInfo));
 
-pub(crate) struct WindowData<T: 'static> {
+/// some backends like macos uses an uninhabited `Never` type,
+/// on windows, `UserEvent`s are also dispatched through the
+/// WNDPROC callback, and due to the re-entrant nature of the
+/// callback, recursively delivered events must be queued in a
+/// buffer, the current implementation put this queue in
+/// `EventLoopRunner`, which is shared between the event pumping
+/// loop and the callback. because it's hard to decide from the
+/// outside whether a event needs to be buffered, I decided not
+/// use `Event<Never>` for the shared runner state, but use unit
+/// as a placeholder so user events can be buffered as usual,
+/// the real `UserEvent` is pulled from the mpsc channel directly
+/// when the placeholder event is delivered to the event handler
+type UserEventPlaceholder = ();
+
+// here below, the generic `EventLoopRunnerShared<T>` is replaced with
+// `EventLoopRunnerShared<UserEventPlaceholder>` so we can get rid
+// of the generic parameter T in types which don't depend on T.
+// this is the approach which requires minimum changes to current
+// backend implementation. it should be considered transitional
+// and should be refactored and cleaned up eventually, I hope.
+
+pub(crate) struct WindowData {
     pub window_state: Arc<Mutex<WindowState>>,
-    pub event_loop_runner: EventLoopRunnerShared<T>,
+    pub event_loop_runner: EventLoopRunnerShared<UserEventPlaceholder>,
```

Comment:
It seems like every use of `EventLoopRunner` is now parametrized by `UserEventPlaceholder`, maybe you can move the generic into `EventLoopRunner::event_handler`? But unsure about this, might be better to keep it minimal.

Reply:
in fact that's the first step in my initial proof of concept attempt, to remove the generic parameter T on `EventLoopRunner`. then not only all the uses of `EventLoopRunner` need to change, so does the implementation in file `platform_impl/windows/event_loop/runner.rs`, and a lot more diff lines were produced.

I decided to make code change at a minimal just to be safe.